pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -384,6 +409,52 @@ func (c *defaultControllerInstallationControl) isResponsible(controllerInstallat
 	return false, nil
 }
 
+func (c *defaultControllerInstallationControl) cleanOldExtensionKinds(ctx context.Context, seedClient client.Client, controllerRegistration *gardencorev1alpha1.ControllerRegistration) (bool, error) {
+	var fns []flow.TaskFn
+
+	extensionGroupVerions := kubernetes.SeedScheme.PrioritizedVersionsForGroup(gardenextensions.GroupName)
+	for _, res := range controllerRegistration.Spec.Resources {
+		for _, gv := range extensionGroupVerions {
```

Comment:
Although the logic here is correct, I'm hesitating a little.. should we first do it only for `Extension` resources? An accidental deletion of a provider extension might lead to destruction of entire shoot clusters. WDYT?

Reply:
Sure, sound reasonable.